### PR TITLE
Fix error by two pivot with one source #1195

### LINKED ISSUE #1195: The modified ClosedXML file opens in Excel with an error.
**Do you want to request a *feature* or report a *bug*?**
- [x] Bug

**Version of ClosedXML**

e.g. 0.95.0-beta1

**Did this work in previous versions of our tool?  Which versions?**

No

## Reproducibility
I created in MS Excel a file containing two Pivot on different sheets, the data source for which is a table with data. Then I opened this file in ClosedXML and changed the data in the data table and saved the file. When opening the received file, MS Excel displays an error:
Deleted component: Component /xl/pivotTables/pivotTable1.xml. (Summary Table View)
Deleted Records: PivotTable Report from Part /xl/workbook.xml (Book)

**Code to reproduce problem:**
see referenced PR

[Failure xlsx-file](https://github.com/ClosedXML/ClosedXML/files/3130851/tPivot5_Static_new.xlsx)

### PR DESCRIPTION
**Pull request by b0bi79**
#### What's this PR do?
I created in MS Excel a file containing two Pivot on different sheets, the data source for which is a table with data. Then I opened this file in ClosedXML and changed the data in the data table and saved the file. When opening the received file, MS Excel displays an error:
Deleted component: Component /xl/pivotTables/pivotTable1.xml. (Summary Table View)
Deleted Records: PivotTable Report from Part /xl/workbook.xml (Book)

The reason that in the pivotCacheDefinition the TaxRate field was not filled in was that the file was generated based on a template in which two summary tables are constructed from data from the same data table. In this case Excel creates one pivotCacheDefinition file, and ClosedXML while caches the last Pivot overwrites the cache for the first Pivot.

#### How should this be manually tested?
The test ClosedXML_Tests.XLPivotTableTests.TwoPivotWithOneSourceTest()

Fixes #1195

### CODE REVIEW

File: ClosedXML/Excel/XLWorkbook_Save.cs [hunk 1]
```diff
@@ -2366,10 +2392,13 @@ private static void GenerateTableDefinitionPartContent(TableDefinitionPart table
             };
 
             if (pt.ReportFilters.Any())
+            {
                 // Reference cell is the part BELOW the report filters
-                location.Reference = pt.TargetCell.CellBelow(pt.ReportFilters.Count() + 1).Address.ToString();
+                var targetCell = pt.TargetCell.CellBelow(pt.ReportFilters.Count() + 1);
```
[the remark this answers is not in the document]
Comment (b0bi79) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: In some cases, when I open the generated file in Excel, I get an error if location.Reference points to a single cell. By changing this to a range of cells, the error goes away.

File: ClosedXML_Tests/Excel/PivotTables/XLPivotTableTests.cs [hunk 2]
```diff
@@ -655,6 +655,24 @@ public void MaintainPivotTableIntegrityOnMultipleSaves()
             }
         }
 
+        [Test]
+        public void TwoPivotWithOneSourceTest()
+        {
+            using (var stream = TestHelper.GetStreamFromResource(TestHelper.GetResourcePath(@"Examples\PivotTables\TwoPivot1195_template.xlsx")))
```
[the remark this answers is not in the document]
Comment (b0bi79) [reply ClosedXML_Tests/Excel/PivotTables/XLPivotTableTests.cs]: done

File: ClosedXML_Tests/Excel/PivotTables/XLPivotTableTests.cs [hunk 3]
```diff
@@ -655,6 +655,24 @@ public void MaintainPivotTableIntegrityOnMultipleSaves()
...
+                var wb = new XLWorkbook(stream);
+                var srcRange = wb.Range("Sheet1!$B$2:$H$207");
+
+                foreach (var pt in wb.Worksheets.SelectMany(ws => ws.PivotTables))
+                {
+                    pt.SourceRange = srcRange;
+                }
+
+                return wb;
+            }, @"Examples\PivotTables\TwoPivot1195.xlsx");
```
[the remark this answers is not in the document]
Comment (b0bi79) [reply ClosedXML_Tests/Excel/PivotTables/XLPivotTableTests.cs]: done

File: ClosedXML/Excel/XLWorkbook_Save.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (igitur) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: You'll have to show me an example for that please.

[the remark this answers is not in the document]
Comment (b0bi79) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: Will the broken file that created ClosedXML suffice?

[the remark this answers is not in the document]
Comment (igitur) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: Yes, anything that shows the problem is fine.

[the remark this answers is not in the document]
Comment (b0bi79) [reply ClosedXML/Excel/XLWorkbook_Save.cs]: Sorry, I can not repeat it anymore. Perhaps the original template was strange.

**Comment by igitur** [comment]
@b0bi79 You have to revert the old files in `ClosedXMLL_Tests/Resource/Other/PivotTableReferenceFiles/*` too for the tests to pass.

**Comment by igitur** [comment]
@b0bi79 Can you please give me access to push some changes to you `b0bi79/two_pivot_fix_1195` branch?

**Comment by igitur** [comment]
To do that, I think you have to check this option:
image

**Comment by b0bi79** [comment]
This option is on. Now I will give full access to the repository.

**Comment by b0bi79** [comment]
You are now an collaborator.

**Comment by igitur** [comment]
Thanks for the contribution. I made some additional changes to avoid having to clone elements. It's becoming clear that we'll need a big refactor of pivot tables to allow references to shared pivot sources. But for now, I think this PR is fine. Do you agree with my additional changes?

**Comment by b0bi79** [comment]
Thank you, Francois. I checked it with my tests, all works.

**Comment by igitur** [comment]
@Pankraty Would you mind doing a quick check on this PR too? It will soon be refactored anyway.

commit history for this PR will be squashed.

**Pull request merged**